pkg/cmd/server/origin/master.go

```diff
@@ -380,6 +382,34 @@ func initAPIVersionRoute(root *restful.WebService, prefix string, versions ...st
 		Consumes(restful.MIME_JSON))
 }
 
+// initReadinessCheckRoute initializes an HTTP endpoint for readiness checking
+func initReadinessCheckRoute(root *restful.WebService, path string, readyFunc func() bool, client *osclient.Client) {
+	root.Route(root.GET(path).To(func(req *restful.Request, resp *restful.Response) {
+		var healthzResponse kapi.Status
+		err := client.Get().AbsPath("/healthz").Do().Into(healthzResponse)
```

Comment:
Sure. The Do() returns only the body of the response, not the header, so to parse it we just have two bytes - "ok" instead of also containing the header. Unclear what it contains for messages without a body.